ic-os/components/setupos-scripts/test-setupos.sh

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+CHECK_NETWORK_SCRIPT="${1:-./check-network.sh}"
+CHECK_HARDWARE_SCRIPT="${2:-./check-hardware.sh}"
+
+# ------------------------------------------------------------------------------
+# Override "source" for test environment.
+# ------------------------------------------------------------------------------
+function source() {
+    if [[ "$1" == "/opt/ic/bin/config.sh" || "$1" == "/opt/ic/bin/functions.sh" ]]; then
+        echo "MOCKED: ignoring source of '$1' (file not present in test environment)"
+        return
+    fi
+    builtin source "$1"
+}
+
+# ------------------------------------------------------------------------------
+# Mocked Functions
+# ------------------------------------------------------------------------------
+
+function log_and_halt_installation_on_error() {
+    if [ "$1" != "0" ]; then
+        echo "ERROR encountered: $2"
+        exit 1
+    fi
+}
+
+# ------------------------------------------------------------------------------
+# Unit tests for check-network.sh
+# ------------------------------------------------------------------------------
+
+function test_validate_domain_name() {
+    echo "Running test: test_validate_domain_name_valid"
+    domain_name="example.com"
+    if validate_domain_name; then
+        echo "  PASS: valid domain: $domain_name"
+    else
+        echo "  FAIL: valid domain: $domain_name"
```

Comment:
Maybe add a more descriptive error message
```suggestion
        echo "  FAIL: failed domain validation, while the domain is valid: $domain_name"
```